container/lxd/server.go

```diff
@@ -63,6 +63,16 @@ func NewLocalServer() (*Server, error) {
 	return svr, errors.Trace(err)
 }
 
+// NewRemoteServer returns a Server based on a remote connection.
+func NewRemoteServer(remote RemoteServer) (*Server, error) {
+	cSvr, err := ConnectRemote(remote)
```

Comment:
Here, where we know for sure that we are connecting to the remove and then using the connection to create a server, we can prevent a double request by doing:
```
remote.SkipGetServer = true
```
As this is executing first thing in NewServer.